sematic/ui/src/components/GitInfo.tsx

```diff
@@ -0,0 +1,128 @@
+import { useCallback, useState } from "react";
+import { Box, ButtonBase, Link, Tooltip, Typography, useTheme } from "@mui/material";
+import { ContentCopy, OpenInNew } from "@mui/icons-material";
+import PostAddIcon from '@mui/icons-material/PostAdd';
+import { RiGitBranchLine, RiGitCommitLine, RiGitRepositoryCommitsLine } from "react-icons/ri";
+import { FaGitSquare } from "react-icons/fa";
+import { Resolution } from "../Models";
+
+/**
+ * Turns the following remote formats:
+ * - git@gihub.com:sematic-ai/sematic.git
+ * - https://github.com/sematic-ai/sematic.git
+ * into clickable links:
+ * - https://github.com/sematic-ai/sematic/tree/<branch>
+ * - https://github.com/sematic-ai/sematic/commit/<SHA>
+ */
+function makeGithubLink(remote: string, path: string) {
+  let domain = remote
+    .replace(/^(git@)|(https:\/\/)/, "")
+    .replace(/(\.git)$/, "")
+    .replace(/:/, "/")
+  return "https://" + domain + "/" + path
+}
+
+function GitInfo(props: {
+    text: string, tooltip: string, remote: string, path: string, children?: any
+}) {
+
+  const [content, setContent] = useState(props.text);
```

Comment:
One common pattern to unpack props is the following:
```typescript
function GitInfo(props: {
    text: string, tooltip: string, remote: string, path: string, children?: any
}) {
  const { text, tooltip, remote, path, children } = props;

  const [content, setContent] = useState(text); // then you can use individual values directly
```

Not strictly required here, just a tip.

Reply:
Done.